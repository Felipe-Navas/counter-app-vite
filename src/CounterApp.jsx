import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value)

  const handleAdd = (event) => {
    // console.log(event)

    setCounter(counter + 1)
    // * Other way to do it:
    // setCounter((c) => c + 1)
  }

  const handleReset = () => {
    setCounter(value)
  }

  const handleSubtract = () => {
    setCounter(counter - 1)
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>

      {/* <button onClick={ (event) => handleAdd(event) }>+1</button> */}
      <button onClick={ handleAdd }>+1</button>
      <button onClick={ handleReset }>Reset</button>
      <button aria-label="btn-reset" onClick={ handleSubtract }>-1</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}
