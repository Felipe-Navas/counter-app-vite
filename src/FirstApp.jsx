import PropTypes from 'prop-types'

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>
      <h2>{subTitle}</h2>
      <h3>{name}</h3>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia
        repudiandae eaque animi saepe. Quidem corporis architecto dicta
        consequuntur eaque consequatur numquam iste itaque cupiditate odio
        repudiandae, ab ipsa eveniet ducimus.
      </p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'Felipe Navas',
  subTitle: 'No subtitle',
  // title: 'No title',
}
