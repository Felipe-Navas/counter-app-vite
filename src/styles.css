
html, body {
  background-color: #21232C;
  color: #fff;
  font-family: 'Roboto', sans-serif;
  font-size: 16px;
  padding: 70px;
}

button {
  padding: 15px;
  font-size: 16px;
  margin-right: 5px;
}