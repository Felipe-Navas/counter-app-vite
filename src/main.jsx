import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
// import { App } from './App'
import { FirstApp } from './FirstApp'

import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <App /> */}
    <FirstApp title="First app title" />
    {/* <CounterApp value={10} /> */}
  </React.StrictMode>
)
