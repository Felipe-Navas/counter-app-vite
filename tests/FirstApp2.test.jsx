import { render, screen } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'

describe('Testing the FirstApp', () => {
  const title = 'First app title'
  const subTitle = 'First app subtitle'

  test('should render correctly with the snapshot', () => {
    const { container } = render(<FirstApp title={title} />)

    expect(container).toMatchSnapshot()
  })

  test('should show the title in the screen', () => {
    render(<FirstApp title={title} />)

    expect(screen.getByText(title)).toBeTruthy()
   })

   test('should show the title in a h1 tag', () => {
    render(<FirstApp title={title} />)

    expect(screen.getByRole('heading', {level: 1}).innerHTML).toBe(title)
   })

   test('should show the subtitle', () => {
    render(<FirstApp title={title} subTitle={subTitle} />)

    expect(screen.getAllByText(subTitle).length).toBe(1)
   })
})
