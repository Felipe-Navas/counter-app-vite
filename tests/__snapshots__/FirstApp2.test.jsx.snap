// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the FirstApp should render correctly with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    First app title
  </h1>
  <h2>
    No subtitle
  </h2>
  <h3>
    Felipe Navas
  </h3>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia repudiandae eaque animi saepe. Quidem corporis architecto dicta consequuntur eaque consequatur numquam iste itaque cupiditate odio repudiandae, ab ipsa eveniet ducimus.
  </p>
</div>
`;
